pkg/driver/controller_modify_volume.go

```diff
@@ -107,55 +113,106 @@ func mergeModifyVolumeRequest(input modifyVolumeRequest, existing modifyVolumeRe
 		}
 		existing.modifyDiskOptions.VolumeType = input.modifyDiskOptions.VolumeType
 	}
-
+	if input.modifyTagsOptions != nil {
+		if len(input.modifyTagsOptions.TagsToAdd) > 0 || len(input.modifyTagsOptions.TagsToDelete) > 0 {
+			if existing.modifyTagsOptions != nil && input.modifyTagsOptions != existing.modifyTagsOptions {
+				return existing, fmt.Errorf("Different tags were requested by a previous request. Current: %v, Requested: %v", existing.modifyTagsOptions, input.modifyTagsOptions)
+			}
+			existing.modifyTagsOptions = &cloud.ModifyTagsOptions{
+				TagsToAdd:    input.modifyTagsOptions.TagsToAdd,
+				TagsToDelete: input.modifyTagsOptions.TagsToDelete,
+			}
+		}
+	}
 	return existing, nil
 }
 
+func executeModifyTagsRequest(volumeID string, options modifyVolumeRequest, c cloud.Cloud) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+	if options.modifyTagsOptions != nil {
+		err := c.ModifyTags(ctx, volumeID, options.modifyTagsOptions.TagsToAdd, options.modifyTagsOptions.TagsToDelete)
+		if err != nil {
+			if errors.Is(err, cloud.ErrInvalidArgument) {
+				return status.Errorf(codes.InvalidArgument, "Could not modify volume tags (invalid argument) %q: %v", volumeID, err)
```

Comment:
Please use `%w` instead of `%v` for errors in format strings. (Even if other parts of codebase use `%v`)

E.g.

`return status.Errorf(codes.Internal, "Could not modify volume tags %q: %w", volumeID, err)`

This is important because it allows the original error to be "wrapped" by the new error, creating a tree of errors, so that later the error can be "unwrapped" and compared with [errors.Is](https://golang.org/pkg/errors/#Is) and [errors.As](https://golang.org/pkg/errors/#As).

See this StackOverflow answer for more info on preferring %w over %v: https://stackoverflow.com/a/61287626

See [Errors GoDoc](https://pkg.go.dev/errors#pkg-overview) for an example of wrapping and unwrapping errors.

Reply:
Edit: I was wrong, gRPC status.Errorf does not let you use `%w`. Please ignore this comment in the case of `status.Errorf`. 

Leads to ugly formatting.